### DISCUSSION TITLE
decompressSync only returning partial data

### DISCUSSION THREAD

**Discussion by jayanthanish** [101arrowz/fflate#178] [General]
We are getting a compressed base64 encoded data from the server and we are passing that data as input in the below code.

const encodeString = "<data received from backend>";
const decodedString = atob(encodedString); 
const decompressed = fflate.decompressSync(
fflate.strToU8(decodedString, true));
const origText = fflate.strFromU8(decompressed);
console.log(origText); 

But the output that we got from the above code is only partial data.
We are migrating from pako, when we pass the same data in pako, we are getting complete data.

Please guide us.

**Comment by 101arrowz** [comment]
If this is concatenated GZIP data, you'll need to parse this with the streaming API rather than the function-based API (which supports only single-entry GZIP files).

Replace your `const decompressed = fflate.decompressSync(fflate.strToU8(decodedString, true));` line with this:

```js
let origText = '';
const textDecoder = new fflate.DecodeUTF8((str, final) => {
  origText += str;
});
const decompressor = new fflate.Decompress((chunk, final) => {
  textDecoder.push(chunk, final);
});
decompressor.push(fflate.strToU8(decodedString, true), true);
console.log(origText);
```

If that doesn't work, please send some data so I can reproduce.

**Reply by jayanthanish** [reply]
Hi,
I have replaced the code like you suggested, now I am getting the following error: ' Error: stream finished'.

code:

const encodeString = "<data>";
const decodedString = atob(encodedString); 
let origText = '';
const textDecoder = new fflate.DecodeUTF8((str, final) => {
  origText += str;
});
const decompressor = new fflate.Decompress((chunk, final) => {
  textDecoder.push(chunk, final);
});
decompressor.push(fflate.strToU8(decodedString, true), true);
console.log(origText);

Please guide us.
Thanks.

**Comment by jayanthanish** [comment]
Hi

Please find the attached documents for the code that we are using, it has the encoded string data as well and the Expected output for the encoded data.
When we initially used decompressSync, our output has only partial data and when we used the code that you suggested then we are getting the error: 'stream finished.'

We are using the following url for executing our code: https://npm.runkit.com/fflate
Please guide us.
Thanks.

[Expected output.txt](https://github.com/101arrowz/fflate/files/12275180/Expected.output.txt)
[Fflate code.txt](https://github.com/101arrowz/fflate/files/12275182/Fflate.code.txt)
